src/compiler/crystal/semantic/ast.cr

```diff
@@ -30,6 +30,58 @@ module Crystal
         false
       end
     end
+
+    # `number_autocast` defines if the feature in #9565 is present
```

Comment:
```suggestion
    # `number_autocast` defines if automatic upcasting for number literals is enabled
```

Reply:
Note that this is not just about number literals, but expressions.

Reply:
Right. "automatic upcasting for number values" then.